syntax = "proto3";

service OrderService {
  rpc CreateOrder(OrderRequest) returns (OrderResponse);
  rpc UpdateOrderStatus(OrderStatusUpdate) returns (OrderResponse);
}

message OrderRequest {
  string product_name = 1;
  float price = 2;
  string payment_gateway = 3;
  string card_brand = 4;
  string bank = 5;
  string region = 6;
  string address = 7;
  string email = 8;
}

message OrderStatusUpdate {
  int32 order_id = 1;
  string status = 2;
}

message OrderResponse {
  string message = 1;
  int32 order_id = 2;
}
